akka-actor/src/main/scala/akka/pattern/BackoffOptions.scala

```diff
@@ -557,27 +567,31 @@ trait BackoffOptions {
    */
   def withReplyWhileStopped(replyWhileStopped: Any): BackoffOptions
 
+  // for backwards compability with 2.5.19
+  @deprecated("Use through BackoffOnStopOptions instead of BackoffOptions", since = "2.5.20")
+  def withFinalStopMessage(isFinalStopMessage: Any ⇒ Boolean): BackoffOptions
+
+  /**
+   * Returns the props to create the back-off supervisor.
+   */
+  private[akka] def props: Props
+}
+
+sealed trait BackoffOnStopOptions extends BackoffOptions {
+
   /**
    * Predicate evaluated for each message, if it returns true and the supervised actor is
    * stopped then the supervisor will stop its self. If it returns true while
    * the supervised actor is running then it will be forwarded to the supervised actor and
    * when the supervised actor stops its self the supervisor will stop its self.
    */
   def withFinalStopMessage(isFinalStopMessage: Any ⇒ Boolean): BackoffOptions
```

Comment:
Return `BackoffOnStopOptions` so that stays within that type for fluent api.

Not important now since there is only one of them but might be more in future.

To be completely nice all in BackoffOptions should be overridden here to return BackoffOnStopOptions, otherwise it's not possible to do `withMaxNrOfRetries(5).withFinalStopMessage(...)

hmm, it kind of explodes

Reply:
Right. Using `this.type` as result type on all `BackoffOptions` method would work in Scala, but not when using from Java. To keep the API the same we'd need to re-define all methods on both sub-traits with the correct result type.